src/WorkflowUpdate/WorkflowUpdate.workflow.cs

```diff
@@ -0,0 +1,65 @@
+namespace TemporalioSamples.WorkflowUpdate;
+
+using Temporalio.Workflows;
+
+[Workflow]
+public class WorkflowUpdate
+{
+    private bool updateInProgress;
+    private ScreenId currentScreen = ScreenId.Screen1;
+
+    [WorkflowRun]
+    public async Task RunAsync() => await Workflow.WaitConditionAsync(() => currentScreen == ScreenId.End);
```

Comment:
```suggestion
    public Task RunAsync() => Workflow.WaitConditionAsync(() => currentScreen == ScreenId.End);
```

Probably could be this, but what you have is good too